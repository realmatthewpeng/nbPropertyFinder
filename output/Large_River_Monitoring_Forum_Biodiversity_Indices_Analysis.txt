Total number of lines found: 128

last = len(data.columns)
data["Margalef"] = ""
data.loc[i,'Margalef'] = (len(data.iloc[i][4:last][data.iloc[i][4:last]>0])-1)/math.log(sum(data.iloc[i][4:last][data.iloc[i][4:last]>0]))
segments = data.RiverSeg.unique()
river = data.iloc[0]["River"]
x = segDF["Year"]
segDF = segDF.sort_values('Year')
segDF = segDF[segDF["Margalef"]>0]
y = segDF["Margalef"]
segDF = data.loc[data['RiverSeg'] == segment]
data["SWI_2"] = ""
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])),2)
data.loc[i,'SWI_2'] = swi
segDF = segDF[segDF["SWI_2"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["SWI_2"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["SWI_e"] = ""
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])))
data.loc[i,'SWI_e'] = swi
segDF = segDF[segDF["SWI_e"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["SWI_e"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["SWI_10"] = ""
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])),10)
data.loc[i,'SWI_10'] = swi
segDF = segDF[segDF["SWI_10"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["SWI_10"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["lam"] = ""
data.loc[i,'lam'] = lam
y = segDF["lam"]
segDF = data.loc[data['RiverSeg'] == segment]
segDF = segDF[segDF["lam"]>0]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["1-lam"] = ""
data.loc[i,'1-lam'] = 1 - lam
segDF = segDF[segDF["1-lam"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["1-lam"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["lam'"] = ""
array = data.iloc[i][4:last]
data.loc[i,"lam'"] = lam
segDF = segDF[segDF["lam'"]>0]
y = segDF["lam'"]
segDF = data.loc[data['RiverSeg'] == segment]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["1-lam'"] = ""
array = data.iloc[i][4:last]
data.loc[i,"1-lam'"] = 1 - lam
y = segDF["1-lam'"]
segDF = segDF[segDF["lam'"]>0]
x = segDF["Year"]
segDF = segDF.sort_values('Year')
segDF = data.loc[data['RiverSeg'] == segment]
data["N1"] = ""
N1 += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])))
data.loc[i,'N1'] = math.exp(N1)
segDF = segDF[segDF["lam'"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["N1"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["N2"] = ""
data.loc[i,'N2'] = 1/lam
segDF = segDF[segDF["lam'"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["N2"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["N_Inf"] = ""
array = data.iloc[i][4:last]
data.loc[i,'N_Inf'] = 1/(max(array)/N)
segDF = segDF[segDF["lam'"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["N_Inf"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["N10"] = ""
data.loc[i,'N10'] = math.exp(swi)/len(data.iloc[i][4:last][data.iloc[i][4:last]>0])
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])))
x = segDF["Year"]
segDF = segDF.sort_values('Year')
segDF = segDF[segDF["lam'"]>0]
y = segDF["N10"]
segDF = data.loc[data['RiverSeg'] == segment]
data["N10'"] = ""
data.loc[i,"N10'"] = 0
data.loc[i,"N10'"] = (math.exp(swi)-1)/(len(data.iloc[i][4:last][data.iloc[i][4:last]>0])-1)
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])))
segDF = segDF[segDF["lam'"]>0]
y = segDF["N10'"]
segDF = data.loc[data['RiverSeg'] == segment]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["N21"] = ""
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])))
data.loc[i,"N21"] = N2/N1
segDF = segDF[segDF["lam'"]>0]
y = segDF["N21"]
segDF = data.loc[data['RiverSeg'] == segment]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["N21'"] = ""
data.loc[i,"N21'"] = (N2-1)/(N1-1)
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])))
segDF = segDF[segDF["lam'"]>0]
y = segDF["N21'"]
segDF = data.loc[data['RiverSeg'] == segment]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
data["Pielou"] = ""
swi += -(data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])) * math.log((data.iloc[i][x]/sum(data.iloc[i][4:last][data.iloc[i][4:last]>0])))
data.loc[i,'Pielou'] = swi/math.log(count)
segDF = segDF[segDF["lam'"]>0]
segDF = data.loc[data['RiverSeg'] == segment]
y = segDF["Pielou"]
segDF = segDF.sort_values('Year')
x = segDF["Year"]

Incorrect Lines:


Precision = 1.0
Recall = 0.9922480620155039

data.iloc[0:5]
