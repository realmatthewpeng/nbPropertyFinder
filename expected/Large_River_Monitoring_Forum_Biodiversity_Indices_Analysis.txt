last = len(data.columns)
data.iloc[0:5]
data["Margalef"] = ""
data.loc[i,'Margalef'] = (len(data.iloc[i][4:last][data.iloc[i][4:last]>0])-1)/math.log(sum(data.iloc[i][4:last][data.iloc[i][4:last]>0]))
segments = data.RiverSeg.unique()
river = data.iloc[0]["River"]
segDF = data.loc[data['RiverSeg'] == segment]
segDF = segDF[segDF["Margalef"]>0]
segDF = segDF.sort_values('Year')
x = segDF["Year"]
y = segDF["Margalef"]
data["SWI_2"] = ""
segDF = segDF[segDF["SWI_2"]>0]
y = segDF["SWI_e"]
data["SWI_10"] = ""
segDF = segDF[segDF["SWI_10"]>0]
y = segDF["SWI_10"]
data["lam"] = ""
segDF = segDF[segDF["lam"]>0]
y = segDF["lam"]
data["1-lam"] = ""
segDF = segDF[segDF["1-lam"]>0]
y = segDF["1-lam"]
data["lam'"] = ""
segDF = segDF[segDF["lam'"]>0]
y = segDF["lam'"]
data["1-lam'"] = ""
y = segDF["1-lam'"]
data["N1"] = ""
y = segDF["N1"]
data["N2"] = ""
y = segDF["N2"]
data["N_Inf"] = ""
y = segDF["N_Inf"]
data["N10"] = ""
y = segDF["N10"]
data["N10'"] = ""
y = segDF["N10'"]
data["N21"] = ""
y = segDF["N21"]
data["N21'"] = ""
y = segDF["N21'"]
data["Pielou"] = ""
y = segDF["Pielou"]